satellite/satellitedb/overlaycache.go

```diff
@@ -909,42 +877,30 @@ func convertDBNode(ctx context.Context, info *dbx.Node) (_ *overlay.NodeDossier,
 
 func getNodeStats(dbNode *dbx.Node) *overlay.NodeStats {
 	nodeStats := &overlay.NodeStats{
-		Latency90:          dbNode.Latency90,
-		AuditSuccessRatio:  dbNode.AuditSuccessRatio,
-		AuditSuccessCount:  dbNode.AuditSuccessCount,
-		AuditCount:         dbNode.TotalAuditCount,
-		UptimeRatio:        dbNode.UptimeRatio,
-		UptimeSuccessCount: dbNode.UptimeSuccessCount,
-		UptimeCount:        dbNode.TotalUptimeCount,
-		LastContactSuccess: dbNode.LastContactSuccess,
-		LastContactFailure: dbNode.LastContactFailure,
+		Latency90:             dbNode.Latency90,
+		AuditCount:            dbNode.TotalAuditCount,
+		UptimeCount:           dbNode.TotalUptimeCount,
+		LastContactSuccess:    dbNode.LastContactSuccess,
+		LastContactFailure:    dbNode.LastContactFailure,
+		AuditReputationAlpha:  dbNode.AuditReputationAlpha,
+		AuditReputationBeta:   dbNode.AuditReputationBeta,
+		UptimeReputationAlpha: dbNode.UptimeReputationAlpha,
+		UptimeReputationBeta:  dbNode.UptimeReputationBeta,
+		Disqualified:          dbNode.Disqualified,
 	}
 	return nodeStats
 }
 
-func updateRatioVars(newStatus bool, successCount, totalCount int64) (int64, int64, float64) {
-	totalCount++
-	if newStatus {
-		successCount++
-	}
-	newRatio := float64(successCount) / float64(totalCount)
-	return successCount, totalCount, newRatio
-}
-
-func checkRatioVars(ctx context.Context, successCount, totalCount int64) (ratio float64, err error) {
-	defer mon.Task()(&ctx)(&err)
-	if successCount < 0 {
-		return 0, errs.New("success count less than 0")
-	}
-	if totalCount < 0 {
-		return 0, errs.New("total count less than 0")
-	}
-	if successCount > totalCount {
-		return 0, errs.New("success count greater than total count")
-	}
-	if totalCount == 0 {
-		return 0, nil
-	}
-	ratio = float64(successCount) / float64(totalCount)
-	return ratio, nil
+// updateReputation uses the Beta distribution model to determine a node's reputation.
+// lambda is the "forgetting factor" which determines how much past info is kept when determining current reputation score.
+// w is the normalization weight that affects how severely new updates affect the current reputation distribution.
+func updateReputation(isSuccess bool, alpha, beta, lambda, w float64, totalCount int64) (newAlpha, newBeta float64, updatedCount int64) {
+	// v is a single feedback value that allows us to update both alpha and beta
+	var v float64 = -1
+	if isSuccess {
+		v = 1
+	}
+	newAlpha = lambda*alpha + w*(1+v)/2
```

Comment:
--> just a thought, thought can it be simplified?
func updateReputation(isSuccess bool, alpha, beta, lambda, w float64, totalCount int64) (newAlpha, newBeta float64, updatedCount int64) {
	// v is a single feedback value that allows us to update both alpha and beta
	if isSuccess {
		// v= 1
		newAlpha = lambda*alpha + w
		newBeta = lambda * beta
	} else {
		//v = -1
		newAlpha = lambda * alpha
		newBeta = lambda*beta + w
	}
	return newAlpha, newBeta, totalCount + 1
}